mono/metadata/threads.c

```diff
@@ -2536,6 +2590,47 @@ mono_thread_resume (MonoInternalThread *thread)
 	return TRUE;
 }
 
+/* LOCKING: LOCK_THREAD(thread) must be held */
+static gboolean
+mono_thread_resume_handle (MonoInternalThreadHandle thread)
```

Comment:
same as `async_abort_internal_handle`: try to have only one copy of the logic and rewrite one static function in terms of the other and leave a FIXME to come back and fixup the callers.